File: go/pkg/models/operations/showpetbyid.go
package operations

import (
	"github.com/speakeasy-sdks-staging/petstore/pkg/models/shared"
)

type ShowPetByIDPathParams struct {
	PetID string `pathParam:"style=simple,explode=false,name=petId"`
}

type ShowPetByIDRequest struct {
	PathParams ShowPetByIDPathParams
}

type ShowPetByIDResponse struct {
	ContentType string
	Error       *shared.Error
	Pet         *shared.Pet
	StatusCode  int
}

File: python/src/sdk/models/operations/showpetbyid.py
import dataclasses
from ..shared import error as shared_error
from ..shared import pet as shared_pet
from typing import Optional


@dataclasses.dataclass
class ShowPetByIDPathParams:
    pet_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'petId', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class ShowPetByIDRequest:
    path_params: ShowPetByIDPathParams = dataclasses.field()
    

@dataclasses.dataclass
class ShowPetByIDResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    error: Optional[shared_error.Error] = dataclasses.field(default=None)
    pet: Optional[shared_pet.Pet] = dataclasses.field(default=None)
    